framework/src/media/media_recorder.c

```diff
@@ -0,0 +1,342 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/media_recorder.h>
+#include <media/media_utils.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+#define IS_SOCKET(fd) (fd > CONFIG_NFILE_DESCRIPTORS)
+#define RECORD_LOCK() do { pthread_mutex_lock(&g_rc.mutex_record); } while (false)
+#define RECORD_UNLOCK() do { pthread_mutex_unlock(&g_rc.mutex_record); } while (false)
+
+struct media_record_context_s {
+	pthread_t           pth;
+	bool                thread_running;
+	pthread_mutexattr_t mutex_attr_record;
+	pthread_mutex_t     mutex_record;
+	sem_t               sem_pause;
+	struct pcm         *pcmin;
+	struct pcm_config  *pcmconf;
+	unsigned int        buffer_size;
+	int                 fd;
+	media_format_e      format;
+	record_state_t      state;
+};
+
+static struct media_record_context_s g_rc;
+static bool b_started = false;
+
+int threadfunc_media_recording(void *args);
+
+/**
+ * Public Functions
+ */
+record_result_t media_record(int fd)
+{
+	pthread_attr_t attr;
+	struct sched_param sparam;
+
+	g_rc.fd = fd;
+
+	if (IS_SOCKET(g_rc.fd)) {
+		char instruction = 0x04;
```

Comment:
Is it generic socket usage?? if it is specific, you can add them into record_config from application.